src/gluonts/model/lstnet/_network.py

```diff
@@ -0,0 +1,269 @@
+            )  # NCT
+            self.cnn.cast(dtype)
+            self.dropout = nn.Dropout(dropout_rate)
+            self.rnn = self._create_rnn_layer(
+                channels, rnn_num_layers, rnn_cell_type, dropout_rate
+            )  # NTC
+            self.rnn.cast(dtype)
+            self.skip_rnn = self._create_rnn_layer(
+                channels, skip_rnn_num_layers, skip_rnn_cell_type, dropout_rate
+            )  # NTC
+            self.skip_rnn.cast(dtype)
+            # TODO: add temporal attention option
+            self.fc = nn.Dense(num_series, dtype=dtype)
+            self.ar_fc = nn.Dense(1, dtype=dtype)
+            if scaling:
+                self.scaler = MeanScaler()
+            else:
+                self.scaler = NOPScaler()
+
+    @staticmethod
+    def _create_rnn_layer(
+        num_cells: int, num_layers: int, cell_type: str, dropout_rate: float
+    ) -> nn.HybridBlock:
+        # TODO: GRUCell activation is fixed to tanh
+        RnnCell = {"lstm": mx.gluon.rnn.LSTMCell, "gru": mx.gluon.rnn.GRUCell}[
+            cell_type
+        ]
+        rnn = mx.gluon.rnn.HybridSequentialRNNCell()
+        for _ in range(num_layers):
+            cell = RnnCell(hidden_size=num_cells)
+            cell = (
+                mx.gluon.rnn.ZoneoutCell(cell, zoneout_states=dropout_rate)
+                if dropout_rate > 0.0
+                else cell
+            )
+            rnn.add(cell)
+        return rnn
+
+    def _skip_rnn_layer(self, F, x: Tensor) -> Tensor:
+        skip_c = F.slice_axis(
+            x, axis=1, begin=-self.channel_skip_count, end=None  # NCT
+        )
+        skip_c = F.reshape(
+            skip_c, shape=(0, 0, -1, self.skip_size)
+        )  # NTCxskip
+        skip_c = F.transpose(skip_c, axes=(0, 3, 1, 2))  # NxskipxTxC
+        skip_c = F.reshape(skip_c, shape=(-3, 0, -1))  # (Nxskip)TC
+        s, _ = self.skip_rnn.unroll(
+            inputs=skip_c,
+            length=min(self.channel_skip_count, self.context_length),
+            layout="NTC",
+            merge_outputs=True,
+            begin_state=self.skip_rnn.begin_state(
+                func=F.zeros,
+                dtype=self.dtype,
+                batch_size=skip_c.shape[0]
+                if isinstance(skip_c, mx.nd.NDArray)
+                else 0,
+            ),
+        )
+        s = F.squeeze(
+            F.slice_axis(s, axis=1, begin=-1, end=None), axis=1
+        )  # (Nxskip)xC
+        s = F.reshape(s, shape=(-1, self.skip_rnn_c_dim))  # Nx(skipxC)
+        return s
+
+    def _ar_highway(self, F, x: Tensor) -> Tensor:
+        ar_x = F.slice_axis(x, axis=2, begin=-self.ar_window, end=None)  # NCT
+        ar_x = F.reshape(ar_x, shape=(-3, 0))  # (NC)xT
+        ar = self.ar_fc(ar_x)
+        ar = F.reshape(ar, shape=(-1, self.num_series))  # NC
+        return ar
+
+    def hybrid_forward(
+        self, F, past_target: Tensor, past_observed_values: Tensor
+    ) -> Tensor:
```

Comment:
It would be better to have a docstring for this method, since its output is used by the derived networks (training and prediction) and it’s not clear what its nature is